### DISCUSSION TITLE
Why Rich Text component cannot bind to field with MySQL varchar data type

### DISCUSSION THREAD

**Discussion by samg666** [Budibase/budibase#3540] [Help] [answered]
Hello,

I have a MySQL varchar filed named `Remark`, and I would like to use Rich Text component to bind this file, but got error `This field setting is the wrong data type for this component`, there's no issue if I use Text Field bind to this field. May I know what kind of data type does the  Rich Text component supported? I tried convert the MySQL field to `text`, still the same error. 

![image](https://user-images.githubusercontent.com/29649651/143422086-1f4013fa-de7b-4772-89c6-883bf9f9edab.png)

**Comment by mjashanks** [comment, marked as answer]
Yea good shout. There's not much that you can do, unfortunately. 

Our Rich Text component is not very good at the moment anyway: https://github.com/Budibase/budibase/issues/2967.

Hope to get this one solved as soon as possible - it's a very important component. I have added a note in this issue to make Rich Text to be usable for any type of Text/String field.

**Reply by samg666** [reply]
thank Michael, hope to have this improved soon.
